### DISCUSSION TITLE
gatsby image generation taking too much time for only 2k images

### DISCUSSION THREAD

**Discussion by yasirhaleem** [gatsbyjs/gatsby#27553] [Help]
Hi,

i have below query, if i add 

`image {
            childImageSharp {
              fixed(width: 380, height: 380, quality: 100) {
                src
              }
            }
          }`
to first query it takes too much time, i would have this in almost all template i have, this means it will take hours to develop, someone told me that doesn't matter how much queries you use, images are cached but in my case, it taking too much longer

```
`export const query = graphql`
  query AllProductsQueries($id: Int!, $category: Int!) {
    strapiProducts(strapiId: { eq: $id }) {
      id
      title  
    slug
      childcategory {
        id
        title
        slug
      }
    }
    allStrapiProducts(filter: { childcategory: { id: { eq: $category } } }) {
      edges {
        node {
          title
          slug
          
          image {
            childImageSharp {
              fixed(width: 380, height: 380, quality: 100) {
                src
              }
            }
          }
          childcategory {
            id
            slug
          }
        }
      }
    }
  }
``
```

**Comment by yasirhaleem** [comment]
p.s i will have almost 12k images

**Comment by LekoArts** [comment]
> someone told me that doesn't matter how much queries you use, images are cached but in my case, it taking too much longer

Well, you have to at least once generate the images, then they are mostly cached (locally). But for builds you'll still generate them every time.

If you plan on having so many images you should look into using e.g. https://www.gatsbyjs.com/cloud/ which can speed up the image generation or a specialized service like Cloudinary. Image processing is time consuming, with those many images you'll need to take some measures to scale this.

**Comment by yasirhaleem** [comment]
Thanks a lot.

so nothing wrong here, it is how it works ? if i'm going to use that much amount of images, its going to take 20-30 minutes on first build ?

**Comment by wardpeet** [comment]
@yasirhaleem it's still early days and we haven't announced this yet but you could try `gatsby-plugin-sharp@lazy-image` for local development, it should skip processing images and only process them when you visit the page.

For builds you'll still have to wait 20-30 minutes. If you're using gatsyby-cloud builds will be faster as we have some optimizations in place.

**Reply by yasirhaleem** [reply]
@wardpeet Thanks for the suggestion, will give it a try.
